be2-scala/src/test/scala/ch/epfl/pop/model/network/method/message/data/federation/ChallengeRequestSuite.scala

```diff
@@ -0,0 +1,24 @@
+package ch.epfl.pop.model.network.method.message.data.federation
+
+import util.examples.Federation.FederationChallengeRequestExample.*
+import ch.epfl.pop.json.MessageDataProtocol.*
+import ch.epfl.pop.model.network.method.message.data.{ActionType, ObjectType}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.funsuite.AnyFunSuite as FunSuite
+import spray.json.*
+import util.examples.Federation.FederationChallengeRequestExample.CHALLENGE_REQUEST
+
+class ChallengeRequestSuite extends FunSuite with Matchers {
+
+  test("Constructor/apply works for RequestChallenge as expected") {
+    CHALLENGE_REQUEST.timestamp should equal(TIMESTAMP)
+    CHALLENGE_REQUEST._object should equal(ObjectType.federation)
+    CHALLENGE_REQUEST.action should equal(ActionType.challenge_request)
+  }
+
+  test("BuildFromJson for RequestChallenge works as expected") {
+    val request_challenge_ = FederationChallengeRequest.buildFromJson(CHALLENGE_REQUEST.toJson.toString)
+
+    request_challenge_ should equal(CHALLENGE_REQUEST)
+  }
```

Comment:
To test that your implementation works with the json examples you could add the json in the /util/examples/json and use readJsonFromPath. That way you can be sure that your implementation is correct with the messages defined in the protocol. Because a subtle bug could be that you are building from a json that you created, so your implementation work to read and write with object within the app, but not for the outside world. You can find some examples in other tests that use readJsonFromPath